app/views/events/_schedule_league.html.erb

```diff
@@ -67,13 +67,17 @@
       <td>
         <%= link_to t('.show', default: t('helpers.links.show')),
                     match_path(match), :class => 'btn btn-default btn-xs' %>
-        <%= link_to t('.edit', default: t('helpers.links.edit')),
-                    edit_match_path(match), :class => 'btn btn-default btn-xs' %>
-        <%= link_to t('.destroy', default: t('helpers.links.destroy')),
-                    match_path(match),
-                    :method => :delete,
-                    :data => { :confirm => t('.confirm', default: t('helpers.links.confirm')) },
-                    :class => 'btn btn-xs btn-danger' %>
+        <% if can? :edit, match %>
+          <%= link_to t('.edit', default: t('helpers.links.edit')),
+                      edit_match_path(match), :class => 'btn btn-default btn-xs' %>
+        <% end %>
+        <% if current_user&.admin %>
```

Comment:
This check should be moved to the abilities file, so that you can use `if can? :destroy, match` here.